pkg/ipamd/ipamd.go

```diff
@@ -1947,7 +1944,7 @@ func (c *IPAMContext) AnnotatePod(podName, podNamespace, key, val string) error
 	var err error
 
 	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		pod := &corev1.Pod{}
+		var pod *corev1.Pod
```

Comment:
Why the change here?

Reply:
This was suggested to change by go-staticcheck as `this value of pod is never used`.

Reply:
I can add a nil check for the pointer if that is the concern.

Reply:
Interesting. We're using it here though - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/ipamd/ipamd.go#L1955 .

Reply:
I would think it was because this `pod, err = c.GetPod(podName, podNamespace)`